local-storage/src/block.rs

```diff
@@ -0,0 +1,263 @@
+use anyhow::{bail, Result};
+use bytes::Bytes;
+use cid::Cid;
+use iroh_unixfs::Block;
+use std::collections::HashSet;
```

Comment:
That looks interesting but appears to be targeted at concurrent access. My current usage of the `HashSet` in this file _should_ always be single-threaded. Will be good to keep dashmap in mind for future set usage though!

Reply:
Is synchronous processing required (i.e. is ordering important) and/or are we assuming a single core setup?

Reply:
In that particular instance I don't think synchronous processing was required, but the operation was so simple (make two sets and compare them once) that it would be an unnecessary amount of extra work to split across threads. However the code in question is being done away with in favor of Chinmay's verification function.